transactions/pendingtxtracker_test.go

```diff
@@ -265,17 +265,29 @@ func TestPendingTxTracker_MultipleClients(t *testing.T) {
 		(*res)["blockNumber"] = transactionSuccessStatus
 	})
 
-	for i := range txs {
-		err := m.TrackPendingTransaction(txs[i].ChainID, txs[i].Hash, txs[i].From, txs[i].Type, true)
-		require.NoError(t, err)
-	}
+	// If we call TrackPendingTransaction immediately, there is a chance that some events
+	// will be emitted before we reach select, so occasionally select fails on timeout.
+	go func() {
+		for i := range txs {
+			err := m.TrackPendingTransaction(txs[i].ChainID, txs[i].Hash, txs[i].From, txs[i].Type, true)
+			require.NoError(t, err)
+		}
+	}()
 
 	eventChan := make(chan walletevent.Event)
 	sub := eventFeed.Subscribe(eventChan)
 
-	err := m.Start()
-	require.NoError(t, err)
+	// events caused by addPending
+	for i := 0; i < 2; i++ {
```

Comment:
It makes sense to have this only if we were to have emitted more events than we processed. As stated in the first comment we are on the other side where we emit fewer events than expected.

Reply:
Now I understand better your changes. However, the last improvements I added to the master branch to subscribe before calling the APIs and increase the channel buffer accordingly is doing the same as you proposed in a different way.